onnxruntime/test/onnx/tensorprotoutils.cc

```diff
@@ -98,7 +98,7 @@ static void UnpackTensorWithRawData(const void* raw_data, size_t raw_data_length
                         OrtErrorCode::ORT_FAIL);                                                          \
     auto& data = tensor.field_name();                                                                     \
     for (auto data_iter = data.cbegin(); data_iter != data.cend(); ++data_iter)                           \
-      *p_data++ = *reinterpret_cast<const T*>(data_iter);                                                 \
+      *p_data++ = (T)(*data_iter);                                                                        \
```

Comment:
Should it be static_cast or gsl::narrow?

Reply:
I think gsl::narrow is better in case if we lose data.

Reply:
> 
1. The casting is needed when datatype is uint8, int8, uint16, int16. Becase in TensorProto, these data_type is stored in int32_data field. 
in TensorProto, 
```
  // For int32, uint8, int8, uint16, int16, bool, and float16 values
  // float16 values must be bit-wise converted to an uint16_t prior
  // to writing to the buffer.
  // When this field is present, the data_type field MUST be
  // INT32, INT16, INT8, UINT16, UINT8, BOOL, FLOAT16 or BFLOAT16
  repeated int32 int32_data = 5 [packed = true];
```
2. gsl::narraw may be a good choice. I will have a try. But here origin reinterpret_cast will cause error.
```
error: invalid cast from type ‘google::protobuf::internal::RepeatedIterator’ to type ‘const uint32_t*’ {aka ‘const unsigned int*’}
101 | *p_data++ = reinterpret_cast<const T>(data_iter);
```